src/components/button/_macro.spec.js

```diff
@@ -5,296 +5,275 @@ import * as cheerio from 'cheerio';
         });
+        describe('WHEN: variants contains loader', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'loader',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-loader')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
+            it('THEN: loader icon when variants contains loader', () => {
+                const faker = templateFaker();
+                const iconsSpy = faker.spy('icon');
 
-        it('has the provided `type` attribute', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    type: 'special-type',
-                }),
-            );
+                faker.renderComponent('button', {
+                    variants: 'loader',
+                });
 
-            expect($('button').attr('type')).toBe('special-type');
+                expect(iconsSpy.occurrences[0].iconType).toBe('loader');
+            });
         });
-
-        it('has the provided `type` attribute even if print variant is provided', () => {
-            const $ = cheerio.load(
-                renderComponent('button', {
-                    type: 'special-type',
-                    variants: 'print',
-                }),
-            );
-
-            expect($('button').attr('type')).toBe('special-type');
+        describe('WHEN: variants contains timer', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'timer',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-js-timer')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-submit-btn')).toBe(true);
+            });
         });
+        describe('WHEN: variants contains print', () => {
+            it('THEN: renders provided variant style classes', () => {
+                const $ = cheerio.load(
+                    renderComponent('button', {
+                        variants: 'print',
+                    }),
+                );
+
+                expect($('.ons-btn').hasClass('ons-btn--print')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-u-d-no')).toBe(true);
+                expect($('.ons-btn').hasClass('ons-js-print-btn')).toBe(true);
+            });
 
-        it('defaults to being a "submit" button when `type` is not provided', () => {
-            const $ = cheerio.load(renderComponent('button'));
-
-            expect($('button').attr('type')).toBe('submit');
-        });
+            it('THEN: renders print icon when variants contains print', () => {
```

Comment:
```suggestion
            it('THEN: button has the print icon', () => {
```

"when variants contains print" is already in the above when